### ISSUE TITLE
For better portability, stop using the type `long`

### ISSUE THREAD

**Issue by ddeclerck** [OCamlPro/gnucobol#159]
A potential source of bugs - and one of the reasons the MSVC debug CIs hangs / displays a runtime checker popup - is the use of the `long` type , whose size differs in 64-bit mode on MSVC (4 bytes) vs "rest of the world" compilers (8 bytes).
(nitpicker's corner: okay, there might be a few other exceptions)

As an example, the test `FUNCTION RANDOM` fails here (`intrinsic.c:cob_intr_random`) :
```
seed = get_seconds_past_midnight () * (long)COB_MODULE_PTR;
```

Indeed, a 64-bit pointer won't fit in a 32-bit long, and the MSVC runtime checker will complain.

Wherever the size matters, I'd suggest using the C99 fixed width integer types `(u)int64_t` - providing a custom definition when missing. Or, we could just use `cob_u64_t` / `cob_s64_t`.

Thoughts ?

**Comment by GitMensch** [comment]
This is the wrong place for this "feature request" - please create that upstream.

Obviously:
* the calls to library functions outside of GnuCOBOL cannot be changed
* API (= external visible) changes may only be done in GC4

C99 types are nice - and should be postponed to GC4 with a fallback possibly provided by gnulib.

For the specific sample: a cast / bitwise truncation to cob_u32_t would be fine as well.

**Comment by ddeclerck** [comment]
Indeed, I opened that on SF => [https://sourceforge.net/p/gnucobol/feature-requests/476/](https://sourceforge.net/p/gnucobol/feature-requests/476/).

**Comment by m4tosm4** [comment]
Type «long» type is really... harmfull...

...and, still, present...

...in... GnuCOBOL...

...today! :-)

Why is that?

The lack of... #if ... #then... #else... directives ? :-)

:-)

**Comment by m4tosm4** [comment]
«This is the wrong place for this "feature request" - please create that upstream.» 

Who says? :-)

Keep your shirt ON, Simon! :-)

Those, in power, are, still, to be... awaken!_ :-)

:-)

**Comment by m4tosm4** [comment]
«C99 types are nice - and should be postponed to GC4 with a fallback possibly provided by gnulib.» - Simon, the wrapper

C99 is nice but, unfortunatly, is... addictive...

...and... counterproductive...

...in a C89/90... development! :-)

:-)
